### PR TITLE
router-and-axios

### PR DESCRIPTION
**Pull request by dfcz3322**

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -5,6 +5,9 @@
     <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
     <meta name="theme-color" content="#000000" />
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
```
Comment (alinalaktionova) [inline public/index.html]: code style

File: src/components/hero/hero.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import { MarvelButton } from '../marvelButton/marvelButton';
+
+interface IHeroProps {
+  character: ICharacter;
+}
+
+export class Hero extends React.Component<IHeroProps> {
+    render() {
+      return (
+        <Grid
```
Comment (alinalaktionova) [inline src/components/hero/hero.tsx]: I think eslint has inappropriate config, cause code style is awful

File: src/components/marvelButton/marvelButton.css [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    border-style: solid;
+    content: "";
+    display: block;
+    height: 16px;
+}
+
+.innerfill {
+    background-color: transparent;
+    border-left: 1px solid #fff;
+    border-right: 1px solid #fff;
```
Comment (alinalaktionova) [inline src/components/marvelButton/marvelButton.css:71]: you can concat this with border-left property

File: src/components/marvelButton/marvelButton.css [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+.innerfill > span {
+    color: white;
+}
+
+.cta-btn .innerfill:after, .cta-btn .innerfill:before {
+    border-color: transparent;
+    border-style: solid;
+    border-width: 0 0 16px 16px;
```
Comment (alinalaktionova) [inline src/components/marvelButton/marvelButton.css:89]: create a variable for the same properties to reduce duplication

File: src/components/marvelButton/marvelButton.css [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    content: "";
+    display: block;
+    position: absolute;
+}
+
+.cta-btn.cta-btn--outline.cta-btn--light .innerfill:after, .cta-btn.cta-btn--outline.cta-btn--light .innerfill:before {
+    background-size: 16px 24px;
+    background-image: repeating-linear-gradient(
+-45deg,#fff,#fff 1px,transparent 0,transparent 16px);
```
Comment (alinalaktionova) [inline src/components/marvelButton/marvelButton.css]: the same with code spacing

File: src/components/marvelButton/marvelButton.css [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    transition: none;
+}
+
+.cta-btn .innerfill:after, .cta-btn .innerfill:before {
```
Comment (alinalaktionova) [inline src/components/marvelButton/marvelButton.css]: you have the same part of code above

File: src/components/marvelButton/marvelButton.tsx [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import React from 'react'
+import './marvelButton.css'
+
+interface IMarvelButtonProps {
+  onClick?: (event: React.MouseEvent<HTMLElement>) => any;
+  margin?: string;
```
Comment (alinalaktionova) [inline src/components/marvelButton/marvelButton.tsx]: can't find usage of margin in your component

File: src/intarface/interface.ts [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    modified: string;
+    name: string;
+    resourceURL: string;
+    series: ISeries;
+    stories: IStories;
+    thumbnail: IThumbnail;
+    urls: IUrls[];
+}
+
+export interface IComics {
```
Comment (alinalaktionova) [inline src/intarface/interface.ts]: so many repeatable properties in interfaces and some interfaces are exactly the same, rewrite them

File: src/pages/search.tsx [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+
+export class Search extends React.Component<RouteChildrenProps, ISearchState> {
+    constructor(props: RouteChildrenProps) {
+        super(props);
+        const searchParam = new URLSearchParams(this.props.location.search).get("search");
+        this.state = {characters: [], searchQuery: searchParam ?? ""};
+      }
+
+      getComics() {
```
Comment (alinalaktionova) [inline src/pages/search.tsx]: api requests should be located in the separate file

File: src/pages/search.tsx [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+        axios.get("http://gateway.marvel.com/v1/public/characters", {
```
Comment (alinalaktionova) [inline src/pages/search.tsx]: create baseUrl for axios requests

File: src/pages/search.tsx [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+            params: {
+                apikey: "cffc6e60c4ebf25fdb0c2a3a87cfdc6f",
```
Comment (alinalaktionova) [inline src/pages/search.tsx]: store api keys in .env variables, it's not secure to write them explicitly in code

File: src/pages/search.tsx [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+                ...(this.state.searchQuery.length && {nameStartsWith: this.state.searchQuery})
```
Comment (alinalaktionova) [inline src/pages/search.tsx]: create a variable for the expression to make it more clear

File: src/pages/search.tsx [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+        });
+      }
+      
+    componentDidMount(){
+        this.getComics();
+    }
+    onSearch = (event: React.MouseEvent<HTMLElement>) => {
+        this.props.history.push({
+            pathname: "/",
+            search: `?search=${this.state.searchQuery}`
```
Comment (alinalaktionova) [inline src/pages/search.tsx]: not a good way to pass state values here, you can have bugs with this approach. use method params to pass actual values

File: src/pages/search.tsx [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+        });
+        this.getComics();
+    }
+    
+    onQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        this.setState({searchQuery: event.target.value});
```
Comment (alinalaktionova) [inline src/pages/search.tsx]: you can change query string in lifecycle methods or in onChange handler

**Review by alinalaktionova — CHANGES_REQUESTED** [review]

File: src/components/marvelButton/marvelButton.css (hunk 3 above)
Comment (dfcz3322) [reply src/components/marvelButton/marvelButton.css:71]: I cannot set them in a single statement: 
https://stackoverflow.com/questions/3126995/combining-border-top-border-right-border-left-border-bottom-in-css

File: .prettierrc.json [hunk 15]
```diff
@@ -0,0 +1,7 @@
+{
+    "trailingComma": "es5",
+    "tabWidth": 4,
```
Comment (alinalaktionova) [inline .prettierrc.json:3]: usually 2 is used

File: .prettierrc.json [hunk 16]
```diff
@@ -0,0 +1,7 @@
...
+    "semi": true,
+    "singleQuote": false,
```
Comment (alinalaktionova) [inline .prettierrc.json:5]: singleQuote is better)) than double

File: src/components/searchBar/searchBar.css [hunk 17]
```diff
@@ -1,12 +1,23 @@
...
-    padding-right: 55px;
-    
+    border-radius: 4px;
 }
 .search-sortBy {
-    color:black;
-    font-weight: bold !important; 
-}
\ No newline at end of file
+    color: white !important;
```
Comment (alinalaktionova) [inline src/components/searchBar/searchBar.css:9]: bad practice to use important

File: src/components/searchLogo/searchLogo.tsx [hunk 18]
```diff
@@ -1,11 +1,13 @@
...
 export class SearchLogo extends React.Component {
-    render() {
-      return (
-        <img className="searchLogo--img" src="https://upload.wikimedia.org/wikipedia/commons/thumb/b/b9/Marvel_Logo.svg/2560px-Marvel_Logo.svg.png"></img>
-        )
+    render(): JSX.Element {
+        return (
+            <img
+                className="searchLogo--img"
+                src="https://upload.wikimedia.org/wikipedia/commons/thumb/b/b9/Marvel_Logo.svg/2560px-Marvel_Logo.svg.png"
```
Comment (alinalaktionova) [inline src/components/searchLogo/searchLogo.tsx]: create variable for this

File: src/pages/comics.tsx [hunk 19]
```diff
@@ -1,9 +1,50 @@
...
+
+export class Comics extends React.Component<RouteChildrenProps, IComicsState> {
+    constructor(props: RouteChildrenProps) {
+        super(props);
+        this.state = { comics: [], isLoading: false };
+    }
+
+    getComics(): void {
+        this.setState({ isLoading: true });
+        handleRequest(`characters/${(this.props.match?.params as any).id}/comics`)
```
Comment (alinalaktionova) [inline src/pages/comics.tsx]: `characters/${(this.props.match?.params as any).id}/comics` not a good way to pass params that way

File: src/pages/comics.tsx [hunk 20]
```diff
@@ -1,9 +1,50 @@
...
+            .then((response) => {
+                console.log(response);
+                this.setState({ comics: response.data.data.results });
```
Comment (alinalaktionova) [inline src/pages/comics.tsx]: parse result data in api file and return end result to use it in react component

File: src/pages/search.tsx [hunk 21]
```diff
@@ -1,63 +1,67 @@
...
 
+    onQueryChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+        this.setState({ searchQuery: event.target.value });
+    };
 
-    render() {
+    render(): JSX.Element {
         return (
             <>
+                {this.state.isLoading ? <LoadingOverlay></LoadingOverlay> : null}
```
Comment (alinalaktionova) [inline src/pages/search.tsx]: no need to have closed tag if there is no body, just <LoadingOverlay/>

**Review by alinalaktionova — CHANGES_REQUESTED** [review]

**Review by alinalaktionova — APPROVED** [review]

**Pull request merged**